### PR TITLE
Update release notes prior to 0.4

### PR DESCRIPTION
**Pull request by OwenArnold**

### CODE REVIEW

**Comment by DanNixon** [comment]
Line 6 of `release-notes.rst` should be changed to match the format of previous releases (e.g. https://github.com/scipp/scipp/blob/master/docs/about/release-notes.rst#v03-march-2020).

File: docs/about/release-notes.rst [hunk 1]
```diff
@@ -26,10 +26,12 @@ Features
 * New realign functionality.
 * Support for event-filtering.
 * Support for subtraction and addition for (realigned) event data.
+* Non-range slicing changed to preserve coords as attrs rather than dropping
 * ``scipp.neutron``: Instrument view with advanced geometry support, showing correct pixel shapes.
 * Instrument view working on doc pages.
 * Made it simpler to add new ``dtype`` and support ``transform`` for all types.
 * Comparison functions such as ``less``, ``greater_equal``, ...
+* ``all`` and ``any`` can work over all dimensions as well as explicitly provided dimension argument
 * Internal cleanup and documentation additions.
```
Comment (nvaytet) [inline docs/about/release-notes.rst:22]:
```
* Can now convert between Scipp objects and Python dictionaries using ``to_dict`` and ``from_dict``.
* New functions ``collapse`` and ``slices`` can be use to split one or more dimensions of a DataArray to a dict of DataArrays.
```

**Comment by nvaytet** [comment]
I think we also added the `repr` for the scipp module itself, which lists all the Datasets that currently exist in the notebook.

File: docs/about/release-notes.rst [hunk 2]
```diff
@@ -3,35 +3,36 @@
...
 
+Breaking changes since v0.3
```
Comment (nvaytet) [inline docs/about/release-notes.rst:23]:
```suggestion
Breaking changes
```

File: docs/about/release-notes.rst [hunk 3]
```diff
@@ -3,23 +3,9 @@
 Release Notes
 =============
 
-Changes since v0.3
+v0.4 (July 2020)
 ------------------
```
Comment (DanNixon) [inline docs/about/release-notes.rst:7]:
```suggestion
----------------
```

A very minor thing.

**Pull request merged**